engine/src/main/java/org/terasology/recording/EventSystemReplayImpl.java

```diff
@@ -0,0 +1,728 @@
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.AbstractConsumableEvent;
+import org.terasology.entitySystem.event.ConsumableEvent;
+import org.terasology.entitySystem.event.Event;
+import org.terasology.entitySystem.event.EventPriority;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.event.internal.PendingEvent;
+import org.terasology.entitySystem.metadata.EventLibrary;
+import org.terasology.entitySystem.metadata.EventMetadata;
+import org.terasology.entitySystem.systems.ComponentSystem;
+import org.terasology.input.BindAxisEvent;
+import org.terasology.input.BindableButton;
+import org.terasology.input.events.KeyEvent;
+import org.terasology.input.events.MouseButtonEvent;
+import org.terasology.input.events.MouseWheelEvent;
+import org.terasology.logic.characters.CharacterMoveInputEvent;
+import org.terasology.monitoring.PerformanceMonitor;
+import org.terasology.network.BroadcastEvent;
+import org.terasology.network.Client;
+import org.terasology.network.NetworkComponent;
+import org.terasology.network.NetworkEvent;
+import org.terasology.network.NetworkMode;
+import org.terasology.network.NetworkSystem;
+import org.terasology.network.OwnerEvent;
+import org.terasology.network.ServerEvent;
+import org.terasology.world.block.BlockComponent;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.*;
+import java.util.concurrent.BlockingQueue;
+
+
+
+/**
+ * Event System used during a replay. It works almost the same as EventSystemImpl, but it considers the recorded events
+ * and plays them along with allowed game events generated by a normal play session.
+ */
+public class EventSystemReplayImpl implements EventSystem {
+
+    private static final Logger logger = LoggerFactory.getLogger(EventSystemReplayImpl.class);
+
+    private Map<Class<? extends Event>, SetMultimap<Class<? extends Component>, EventSystemReplayImpl.EventHandlerInfo>> componentSpecificHandlers = Maps.newHashMap();
+    private SetMultimap<Class<? extends Event>, EventSystemReplayImpl.EventHandlerInfo> generalHandlers = HashMultimap.create();
+    private Comparator<EventSystemReplayImpl.EventHandlerInfo> priorityComparator = new EventSystemReplayImpl.EventHandlerPriorityComparator();
+
+    // Event metadata
+    private BiMap<SimpleUri, Class<? extends Event>> eventIdMap = HashBiMap.create();
+    private SetMultimap<Class<? extends Event>, Class<? extends Event>> childEvents = HashMultimap.create();
+
+    private Thread mainThread;
+    private BlockingQueue<PendingEvent> pendingEvents = Queues.newLinkedBlockingQueue();
+    private BlockingQueue<RecordedEvent> recordedEvents = Queues.newLinkedBlockingQueue();
+
+    private EventLibrary eventLibrary;
+    private NetworkSystem networkSystem;
+
+    //Event replaying
+    private boolean loadedRecordedEvents; // if the recorded events were loaded from the RecordedEventStore
```

Comment:
Rather than add a `//` style comment after the variable on the same line you can use `/** comment */` style comments on the previous line, and IDEs will be able to use it as javadoc, even if private fields don't get exported to a javadoc site properly

```
/** Whether the recorded events were loaded from the RecordedEventStore */
private boolean loadedRecordedEvents;
```

It also just looks nice :-)